hawc/apps/riskofbias/exports.py

```diff
@@ -1,92 +1,104 @@
-                .select_related("author")
-                .prefetch_related("scores__overridden_objects")
-            )
-            ser = serializers.RiskOfBiasSerializer(qs, many=True)
-            rob_data = ser.data
-            for study in study_data:
-                study["riskofbiases"] = [rob for rob in rob_data if rob["study"] == study["id"]]
-
-        return study_data
-
-    def _get_header_row(self):
-        header = []
-        header.extend(Study.flat_complete_header_row())
-        header.extend(models.RiskOfBias.flat_header_row(final_only=self.final_only))
-        header.extend(models.RiskOfBiasScore.flat_complete_header_row())
-        return header
-
-    def _get_data_rows(self):
-        rows = []
-        for ser in self.get_serialized_data():
-            domains = ser["rob_settings"]["domains"]
-            metrics = ser["rob_settings"]["metrics"]
-            domain_map = {domain["id"]: domain for domain in domains}
-            metric_map = {
-                metric["id"]: dict(metric, domain=domain_map[metric["domain_id"]])
-                for metric in metrics
-            }
-
-            row1 = []
-            row1.extend(Study.flat_complete_data_row(ser))
-
-            robs = [rob for rob in ser.get("riskofbiases", [])]
-            if self.final_only:
-                robs = [rob for rob in robs if rob["final"] and rob["active"]]
-
-            for rob in robs:
-                row2 = list(row1)
-                row2.extend(models.RiskOfBias.flat_data_row(rob, final_only=self.final_only))
-                for score in rob["scores"]:
-                    row3 = list(row2)
-                    score["metric"] = metric_map[score["metric_id"]]
-                    row3.extend(models.RiskOfBiasScore.flat_complete_data_row(score))
-                    rows.append(row3)
-
-        return rows
-
-    def build_metadata(self) -> pd.DataFrame | None:
+from django.db.models import Value
+
+from ..common.exports import Exporter, ModelExport
+from ..common.helper import cleanHTML
+from ..common.models import sql_format
+from ..study.exports import StudyExport
+from . import constants
+
+
+class RiskOfBiasExport(ModelExport):
+    def get_value_map(self):
+        return {
+            "id": "id",
+            "active": "active",
+            "final": "final",
+            "author_id": "author_id",
+            "author_name": "author_full_name",
+            "created": "created",
+            "last_updated": "last_updated",
+        }
+
+    def get_annotation_map(self, query_prefix):
+        return {
+            "author_full_name": sql_format(
+                "{} {}", query_prefix + "author__first_name", query_prefix + "author__last_name"
+            ),
+        }
+
+    def prepare_df(self, df):
+        return self.format_time(df)
+
+
+class DomainExport(ModelExport):
+    def get_value_map(self):
+        return {
+            "domain_id": "id",
```

Comment:
Could you move these prefixes from domain and metric into the exporter? Just for consistency

Reply:
done!